SimTKcommon/src/ParallelExecutor.cpp

```diff
@@ -135,6 +160,9 @@ void* threadBody(void* args) {
             ParallelExecutor::Task& task = executor.getCurrentTask();
             task.initialize();
             int index = info.index;
+            
+            //TODO: Is this the most informative way we can throw an error?
```

Comment:
Since ParallelExecutor has no control over what the tasks are, this might be the best we can do here. The tasks could always output additional messages of their own. To do anything fancier we would need to invent an error reporting mechanism (like promise/future has in C++11). That would then make it harder for people to write tasks.